File: 0461-hamming-distance/0461-hamming-distance.js
/**
 * @param {number} x
 * @param {number} y
 * @return {number}
 */
var hammingDistance = function(x, y) {
    let s = (x ^ y).toString(2); // 转换为二进制字符串
    console.log(s)//101

   let bits = s.split('');     // 按字符切分为数组
   console.log(bits)//[ '1', '0', '1' ]
       let newarray = bits.filter(bit => bit === '1')//
       console.log(newarray)//[ '1', '1' ]
let count = newarray.length
return count
};
//var hammingDistance = (x, y) => (x ^ y).toString(2).split('').filter(b => b === '1').length;

File: 0461-hamming-distance/0461-hamming-distance.cpp
class Solution {
public:
    int hammingDistance(int x, int y) {
          int xorResult = x ^ y; // 先做异或
    int count = 0;

    // 统计 xorResult 中 1 的个数（即不同位的数量）
    while (xorResult != 0) {
        count += xorResult & 1; // 如果最低位是1，加1
        xorResult >>= 1;        // 右移一位
    }

    return count;
}
    
};